Source/WebCore/workers/service/ServiceWorkerJobData.cpp

```diff
@@ -79,8 +94,10 @@ ServiceWorkerJobData ServiceWorkerJobData::isolatedCopy() const
     result.topOrigin = topOrigin.isolatedCopy();
     result.scopeURL = scopeURL.isolatedCopy();
     result.domainForCachePartition = domainForCachePartition.isolatedCopy();
-    result.registrationOptions = registrationOptions.isolatedCopy();
-
+    if (registrationOptions) {
+        ASSERT(type == ServiceWorkerJobType::Register);
+        result.registrationOptions = registrationOptions->isolatedCopy();
```

Comment:
I don't see the need to make this std::optional, but if there is such a need crossThreadCopy can be used with std::optional.
Test failures look related.

Reply:
Test failures were due to my bad validator in the decoder of registrationOptions. That's now fixed.

The reason I made registrationOptions optional is because they're only added when we have ServiceWorkerJobType::Register. It's much easier to both ASSERT this is the case during serialization using a std::optional.

Updated to use crossThreadCopy.